### PR TITLE
build: use clang-9

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

Description:
Risk Level: Med
Testing: CI
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

File: tools/check_format.py [hunk 1]
```diff
@@ -185,8 +185,8 @@ def checkTools():
         "installed, but the binary name is different or it's not available in "
         "PATH, please use CLANG_FORMAT environment variable to specify the path. "
         "Examples:\n"
-        "    export CLANG_FORMAT=clang-format-8.0.0\n"
-        "    export CLANG_FORMAT=/opt/bin/clang-format-8\n"
+        "    export CLANG_FORMAT=clang-format-9.0.0\n"
+        "    export CLANG_FORMAT=/opt/bin/clang-format-9\n"
         "    export CLANG_FORMAT=/usr/local/opt/llvm@8/bin/clang-format".format(CLANG_FORMAT_PATH))
```
Comment (derekargueta) [inline tools/check_format.py]: `llvm@9`

File: ci/run_clang_tidy.sh [hunk 2]
```diff
@@ -46,7 +46,7 @@ function filter_excludes() {
 
 if [[ "${RUN_FULL_CLANG_TIDY}" == 1 ]]; then
   echo "Running full clang-tidy..."
-  run-clang-tidy-8
+  run-clang-tidy-9
 elif [[ -z "${CIRCLE_PR_NUMBER}" && "$CIRCLE_BRANCH" == "master" ]]; then
   echo "On master branch, running clang-tidy-diff against previous commit..."
   git diff HEAD^ | filter_excludes | clang-tidy-diff-8.py -p 1
```
Comment (derekargueta) [inline ci/run_clang_tidy.sh]: `clang-tidy-diff-9.py`?

(and also line 57)

File: bazel/README.md [hunk 3]
```diff
@@ -591,13 +591,13 @@ to run clang-format scripts on your workstation directly:
 
 To run the tools directly, you must install the correct version of clang. This
 may change over time but as of June 2019,
```
Comment (derekargueta) [inline bazel/README.md]: update to `October 2019`?

File: bazel/README.md [hunk 4]
```diff
@@ -145,7 +145,7 @@ bazel build --config=libc++ //source/exe:envoy-static
 ```
 Note: this assumes that both: clang compiler and libc++ library are installed in the system,
 and that `clang` and `clang++` are available in `$PATH`. On some systems, you might need to
-include them in the search path, e.g. `export PATH=/usr/lib/llvm-8/bin:$PATH`.
+include them in the search path, e.g. `export PATH=/usr/lib/llvm-9/bin:$PATH`.
 
 You might also need to ensure libc++ is installed correctly on your system, e.g. on Ubuntu this
 might look like `sudo apt-get install libc++abi-8-dev libc++-8-dev`.
```
Comment (derekargueta) [inline bazel/README.md]: do we want to recommend using v9 of libc++ as well?

File: bazel/README.md (hunk 3 above)
Comment (lizan) [reply bazel/README.md]: yeah I delete the date there.

File: ci/run_clang_tidy.sh (hunk 2 above)
Comment (lizan) [reply ci/run_clang_tidy.sh]: nice catch, thanks

File: bazel/README.md (hunk 4 above)
Comment (lizan) [reply bazel/README.md]: yes

File: .bazelrc [hunk 5]
```diff
@@ -41,8 +41,12 @@ build:asan --define signal_trace=disabled
 build:asan --define ENVOY_CONFIG_ASAN=1
 build:asan --copt -fsanitize=address,undefined
 build:asan --linkopt -fsanitize=address,undefined
+# TODO(lizan): vptr and function requires C++ UBSAN runtime which we're not currently linking to.
+# Enable them when bazel has better support for that.
```
Comment (htuch) [inline .bazelrc]: This seems a pretty major loss of ASAN coverage; is there a reason this only shows up in clang-9? Can you also file a tracking issue?

Comment (lizan) [reply .bazelrc]: Because function ubsan added dependency to RTTI (which requires C++ runtime) https://reviews.llvm.org/D61479.

**Pull request merged**